lib/Ix/DBIC/ResultSet.pm

```diff
@@ -893,4 +893,252 @@ sub ix_set ($self, $ctx, $arg = {}) {
+    # unfortunately, so we get it. -- rjbs, 2016-04-13
+    return $ctx->result("${key1}ListUpdates" => {
+      filter => $arg->{filter},
+      sort   => $arg->{sort},
+      oldState => $since_state,
+      newState => $since_state,
+      total    => $total,
+      removed  => [ ],
+      added    => [ ],
+    });
+  }
+
+  my $filter_map = $rclass->ix_get_list_filter_map;
+  my %required = map {;
+    $_ => $arg->{filter}{$_}
+  } grep {;
+    $filter_map->{$_}->{required}
+  } keys %$filter_map;
+
+  # XXX: stupid, gross, blah -- rjbs, 2016-04-13
+  my @entities = $search->{rs}->search(
+    \%required,
+    $search->{sort}
+  )->all;
+
+  my $i = 0;
+  my @added;
+  my @removed;
+
+  for my $entity (@entities) {
+    my $is_new     = $entity->modSeqCreated > $since_state;
+    my $is_changed = $entity->modSeqChanged > $since_state;
+    my $is_removed = $entity->dateDeleted;
+
+    unless ($is_removed) {
+      FILTER: for my $filter (keys $arg->{filter}->%*) {
+        if ($entity->$filter ne $arg->{filter}{$filter}) {
+          $is_removed = 1;
+          last FILTER;
+        }
+      }
+    }
+
+    my $was_removed = $entity->dateDeleted && ! $is_changed;
+
+    if ($is_removed && ! $is_new && ! $was_removed) {
+      push @removed, "" . $entity->id;
+    } elsif (! $is_removed && $is_new) {
+      push @added, { index => $i, "${key1}Id" => "" . $entity->id };
+    } elsif (! $is_removed && $is_changed) {
+      push @removed, "" . $entity->id;
+      push @added, { index => $i, "${key1}Id" => "" . $entity->id };
+    }
+
+    $i++ unless $is_removed;
+  }
+
+  return $ctx->result("${key1}ListUpdates" => {
+    filter => $arg->{filter},
+    sort   => $arg->{sort},
+    oldState => "" . $since_state,
+    newState => "" . $state->highestModSeq,
+    total    => $total,
+    removed  => \@removed,
+    added    => \@added,
+  });
+}
+
+sub _ml_search_args ($self, $ctx, $arg) {
+  my $rclass = $self->_ix_rclass;
+
+  my %search;
+
+  $search{rs} = $self->search({ 'me.accountId' => $ctx->accountId });
+
+  my %bad_filter;
+
+  my $filter_map = $rclass->ix_get_list_filter_map;
+
+  FILTER: for my $field (keys $arg->{filter}->%*) {
+    if (!$filter_map->{$field}) {
+      $bad_filter{$field} = 'unknown filter field';
+      next FILTER;
+    }
+    my $sql_name = $field;
+    unless ($sql_name =~ /\./) {
+      $sql_name = "me.$sql_name"; # me.<...>
+    }
+
+    my $val = defined $arg->{filter}{$field}
+                ? $arg->{filter}{$field} . ""
+                : undef;
+    $search{filter}{$sql_name} = $val;
+  }
+
+  for my $field (grep {; $filter_map->{$_}->{required} } keys %$filter_map) {
+    unless (exists $arg->{filter}->{$field}) {
+      $bad_filter{$field} = 'required filter missing';
+    }
+  }
+
+  my @sort;
+  my %bad_sort;
+
+  my $sort_map = $rclass->ix_get_list_sort_map;
+
+  SORT: for my $sort ($arg->{sort}->@*) {
+    my ($field, $order) = split / /, $sort;
```

Comment:
I think we need a `, 2` on this.  Otherwise, you could sort "foo desc baz" and the baz is ignored and treated like valid, right?